src/GuiBase/Viewer/Gizmo/ScaleGizmo.cpp

```diff
@@ -0,0 +1,400 @@
+    if ( whole )
+    {
+        auto RO = roMgr->getRenderObject( m_renderObjects[0] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        RO = roMgr->getRenderObject( m_renderObjects[1] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        RO = roMgr->getRenderObject( m_renderObjects[2] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        RO = roMgr->getRenderObject( m_renderObjects[3] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        RO = roMgr->getRenderObject( m_renderObjects[4] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        RO = roMgr->getRenderObject( m_renderObjects[5] );
+        colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+    } else
+    {
+        auto RO = roMgr->getRenderObject( m_renderObjects[0] );
+        colorMesh( RO->getMesh(), Core::Colors::Red() );
+        RO = roMgr->getRenderObject( m_renderObjects[1] );
+        colorMesh( RO->getMesh(), Core::Colors::Green() );
+        RO = roMgr->getRenderObject( m_renderObjects[2] );
+        colorMesh( RO->getMesh(), Core::Colors::Blue() );
+        RO = roMgr->getRenderObject( m_renderObjects[3] );
+        colorMesh( RO->getMesh(), Core::Colors::Red() );
+        RO = roMgr->getRenderObject( m_renderObjects[4] );
+        colorMesh( RO->getMesh(), Core::Colors::Green() );
+        RO = roMgr->getRenderObject( m_renderObjects[5] );
+        colorMesh( RO->getMesh(), Core::Colors::Blue() );
+        if ( m_selectedAxis > 0 )
+        {
+            RO = roMgr->getRenderObject( m_renderObjects[m_selectedAxis] );
+            colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        }
+        if ( m_selectedPlane > 0 )
+        {
+            RO = roMgr->getRenderObject( m_renderObjects[m_selectedPlane + 3] );
+            colorMesh( RO->getMesh(), Core::Colors::Yellow() );
+        }
+    }
+
+    if ( m_selectedAxis == -1 && m_selectedPlane == -1 && !whole )
+    {
+        return m_transform;
+    }
+
+    if ( whole )
+    {
+        const Core::Vector3 origin = m_transform.translation();
+        Core::Vector3 scaleDir = Core::Vector3( m_transform.rotation() * Core::Vector3::Unit( 0 ) );
+
+        if ( !m_start )
+        {
+            if ( findPointOnPlane( cam, origin, scaleDir, m_initialPix + nextXY, m_startPoint ) )
+            {
+                m_start = true;
+                m_startPos = m_transform.translation();
+                m_prevScale = Core::Vector3::Ones();
+            }
+        }
+
+        Core::Vector3 endPoint;
+        if ( findPointOnPlane( cam, origin, scaleDir, m_initialPix + nextXY, endPoint ) )
+        {
+            const Core::Vector3 a = endPoint - m_startPos;
+            if ( a.squaredNorm() < 1e-3 )
+            {
+                return m_transform;
+            }
+            const Core::Vector3 b = m_startPoint - m_startPos;
+            Core::Vector3 scale = a.norm() / b.norm() * Core::Vector3::Ones();
```

Comment:
This constrains the scale to be positive. At use, this results to strange behavior when one might expect to obtain a symmetric object when the scale become negative and see only a kind of "bouncing" object ...

Reply:
updated doc